ddraw/IDirect3DDeviceX.cpp

```diff
@@ -2512,17 +2570,39 @@ void m_IDirect3DDeviceX::InitDevice(DWORD DirectXVersion)
 
 void m_IDirect3DDeviceX::ReleaseDevice()
 {
+
 	WrapperInterface->DeleteMe();
 	WrapperInterface2->DeleteMe();
 	WrapperInterface3->DeleteMe();
 	WrapperInterface7->DeleteMe();
 
+#if WITH_IMGUI
+	if (Config.Dd7to9)
+	{
+		ImGui_ImplDX9_Shutdown();
+		ImGui_ImplWin32_Shutdown();
+		ImGui::DestroyContext();
+	}
+#endif
+
 	if (ddrawParent && !Config.Exiting)
 	{
 		ddrawParent->ClearD3DDevice();
 	}
 }
 
+namespace {
```

Comment:
This block should only compile it IMGUI is enabled. Just try a compile with that disabled and you'll see.

Reply:
Yes, you are right, this would be cleaner. However the code will then be stripped by the linker anyway.

Reply:
I fixed this in check in:  abdbeab8272162b9dbb7bb9d7ec69b4d1f6499cd